rca/enquire_to_study/views.py

```diff
@@ -18,22 +52,112 @@ def dispatch(self, *args, **kwargs):
+            "Notes": f"Enquiry reason: {form_data['enquiry_reason'].reason}",
+            "tags": [
+                {
+                    "TagType": "Source",
+                    "Tag": "rca_wf_xx_RCA Website Webform_xx_xx_xx-xx-xx",
+                }
+            ],
+            "consents": [
+                {
+                    "Consent": form_data["is_read_data_protection_policy"],
+                    "ConsentType": "is_read_data_protection_policy",
+                }
+            ],
+        }
+
+        # USE IOC format to send data, eg CA (Canada) Should be CAN
+        for k, v in IOC_TO_ISO.items():
+            if v == form_data["country_of_citizenship"]:
+                data["CountryOfCitizenship"] = k
+            if v == form_data["country_of_residence"]:
+                data["CountryOfResidence"] = k
+
+        selected_level_of_study_list = []
+        selected_course_list = []
+        for programme in form_data["programmes"]:
+            course_code = next(
+                (
+                    course
+                    for course in qs_courses
+                    if course["codeExternal"] == str(programme.qs_code)
+                ),
+                None,
+            )
+            if not course_code:
+                raise Exception(f"{programme.title} not found in QS course list")
+            course_code = course_code["code"]
+
+            if course_code not in selected_course_list:
+                selected_course_list.append(course_code)
+            level_of_study_code = next(
+                (
+                    level
+                    for level in qs_level_studies
+                    if level["code"] == programme.programme_type.qs_code
+                ),
+                None,
+            )
+        for programme_type in form_data["programme_types"]:
+            level_of_study_code = next(
+                (
+                    level
+                    for level in qs_level_studies
+                    if level["code"] == programme_type.qs_code
+                ),
+                None,
+            )
+            if not level_of_study_code:
+                raise Exception(
+                    f"{programme_type.display_name} not found in QS Level of study list"
+                )
+
+            level_of_study_code = level_of_study_code["code"]
+
+            if level_of_study_code not in selected_level_of_study_list:
+                selected_level_of_study_list.append(level_of_study_code)
+
+        data["LevelOfStudy"] = ",".join(selected_level_of_study_list)
+        data["Course"] = ",".join(selected_course_list)
+        response = requests.post(
+            f"{settings.QS_API_ENDPOINT}/studentEnquiries",
+            json=data,
+            auth=(settings.QS_API_USERNAME, settings.QS_API_PASSWORD),
+        )
+        # If the response was successful, no Exception will be raised
+        response.raise_for_status()
 
     def create_form_submission(self, form):
```

Comment:
Where is `create_form_submission()` called? I wonder if the indirection to call `form.save()` is necessary

Reply:
That's called here https://github.com/torchbox/rca-wagtail-2019/blob/feature/624-qs-integration/rca/enquire_to_study/views.py#L195 and the from.save creates the form submission object. 

Is the suggestion here that I just call `form.save()` [here](https://github.com/torchbox/rca-wagtail-2019/blob/feature/624-qs-integration/rca/enquire_to_study/views.py#L195) instead?